### PR TITLE
[SG-1143] Permissions round 2

### PR DESCRIPTION
**Pull request by zakiya**

### CODE REVIEW

File: web/modules/custom/sfgov_moderation/src/StateTransitionValidation.php [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    if ($entity->getEntityTypeId() != 'node' ||
+      $entity->isNew() ||
+      $user->hasPermission('administer nodes') ||
+      $user->hasPermission('bypass node access') ||
+      in_array($this->publisher(), $user->getRoles())
+    ) {
+      return $validTransitions;
+    }
+
+    // Act on moderated content that belongs to a department.
```
Comment (arlina-espinoza) [inline web/modules/custom/sfgov_moderation/src/StateTransitionValidation.php:143]: We aren't checking departments anymore, correct?

File: web/modules/custom/sfgov_utilities/src/Routing/RouteSubscriberAlter.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function alterRoutes(RouteCollection $collection) {
+
+    // Allow the moderation module to modify access for the node edit form.
+    if ($route = $collection->get('entity.node.edit_form')) {
+      $route->setRequirement('_custom_access', 'sfgov_moderation.access_checker::access');
+    }
```
Comment (arlina-espinoza) [inline web/modules/custom/sfgov_utilities/src/Routing/RouteSubscriberAlter.php:21]: Is this just needed to restrict access to the edit form on translations?

**Review by arlina-espinoza — APPROVED** [review]
I'm so glad this is moving forward. Thanks @zakiya, it looks really good. I just posted a couple of small questions, but I think it's good to merge. We should add a follow up task to remove the group module from the codebase once this is pushed all the way to prod.

File: web/modules/custom/sfgov_utilities/src/Routing/RouteSubscriberAlter.php (hunk 2 above)
Comment (zakiya) [reply web/modules/custom/sfgov_utilities/src/Routing/RouteSubscriberAlter.php:21]: Yes. I thought about naming it more specifically but never got around to it.

File: web/modules/custom/sfgov_moderation/src/StateTransitionValidation.php (hunk 1 above)
Comment (zakiya) [reply web/modules/custom/sfgov_moderation/src/StateTransitionValidation.php:143]: Correct. I'll update the comments. Thanks!

**Review by aekong — APPROVED** [review]
so glad to be rid of the group stuff!

**Pull request merged**